src/subprocess-posix.cc

```diff
@@ -125,25 +126,25 @@ void Subprocess::OnPipeReady() {
 }
 
 ExitStatus Subprocess::Finish() {
-  assert(pid_ != -1);
-  int status;
-  if (waitpid(pid_, &status, 0) < 0)
-    Fatal("waitpid(%d): %s", pid_, strerror(errno));
-  pid_ = -1;
-
-  if (WIFEXITED(status)) {
-    int exit = WEXITSTATUS(status);
+  if (pid_ != -1) {
```

Comment:
Can this happen with the new code? I think previously waitpid() was required here to collect zombies, now we shouldn't have these when Finish() is called.